market/src/main/java/com/banica/market/tasks/TickGeneratorTask.java

```
@@ -0,0 +1,41 @@
+package com.banica.market.tasks;
+
+import com.banica.market.models.GoodSpecification;
+import com.banica.market.models.MarketTick;
+import com.banica.market.utils.TickGenerationUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.BlockingQueue;
+
+public class TickGeneratorTask implements Runnable {
+    private static final Logger logger = LoggerFactory.getLogger(TickGeneratorTask.class);
+
+    private final BlockingQueue<MarketTick> marketTicksQueue;
+    private GoodSpecification goodSpecification;
+    private String origin;
+
+    public TickGeneratorTask(GoodSpecification goodSpecification, String origin, BlockingQueue<MarketTick> marketTicksQueue) {
+        this.goodSpecification = goodSpecification;
+        this.marketTicksQueue = marketTicksQueue;
+        this.origin = origin;
+    }
+
+    @Override
+    public void run() {
+        long amount = TickGenerationUtils.generateRandLongWithStep(goodSpecification.getQuantityLow(),
+                goodSpecification.getQuantityHigh(), goodSpecification.getQuantityStep());
+        double price = TickGenerationUtils.generateRandDoubleWithStep(goodSpecification.getPriceLow(),
+                goodSpecification.getPriceHigh(), goodSpecification.getPriceStep());
+
+        MarketTick marketTick = new MarketTick(origin, goodSpecification.getName(), amount, price);
+        logger.info("Generated " + marketTick);
+
+        try {
+            marketTicksQueue.put(marketTick);
+        } catch (InterruptedException e) {
+            logger.error("Failed to put market tick {} in queue", marketTick);
+            e.printStackTrace();
+        }
+    }
```

Swallowing the interrupted exception may cause problems.
I have a suggestion about adding ticks to the queue, especially if you decide to use bounded queue.
```
if (!marketTicksQueue.offer(marketTick)) {
    logger.warn("The market ticks queue is full. Blocking.");
    try {
        marketTicksQueue.put(marketTick);
    } catch (InterruptedException e) {
        logger.error("Failed to put market tick {} in queue", marketTick);
        Thread.currentThread().interrupt();
    }
}
```
So first try offer, which is non-blocking, but will return false if the queue is full. Then log a warning, so we can notice and adjust the capacity of the queue to a more appropriate value. And only then call put, which would temporarily block the generation of new ticks.